### PR TITLE
[enterprise-4.15] OBSDOCS-1031: Add deprecation info for prometheus-adapter to OCP 4.15…

### PR DESCRIPTION
**Pull request by eromanova97**
Version(s): `enterprise-4.15`

Issue: [OBSDOCS-1031](https://issues.redhat.com/browse/OBSDOCS-1031)

Links to docs preview: 
* [Deprecated and removed features](https://75713--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-15-release-notes#ocp-4-15-deprecated-removed-features)
* [Prometheus Adapter for core platform monitoring](https://75713--ocpdocs-pr.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-15-release-notes#ocp-4-15-prometheus-adapter-for-core-platform-monitoring)

**Additional information:** This issue requires to go through the **change management** process:
- [x] QE has approved this change. (@juzhao)
- [x] Eng has approved this change. (@simonpasquier, @jan--f )
- [x] PM has approved this change. (@w1dg3r)
- [x] PX has approved this change. (@Senthamilarasu-STA)
- [x] DPM has approved this change. (@bburt-rh)

**Edit: The change management has been completed.**

### CODE REVIEW

File: release_notes/ocp-4-15-release-notes.adoc [hunk 1]
```diff
@@ -1431,6 +1436,13 @@ The ability to enable dedicated service monitors by configuring the `dedicatedSe
 To replace this feature, Prometheus functionality has been improved to ensure that alerts and time aggregations are accurate.
 This improved functionality is active by default and makes the dedicated service monitors feature obsolete.
 
+[id="ocp-4-15-prometheus-adapter-for-core-platform-monitoring"]
+==== Prometheus Adapter for core platform monitoring
+With this release, the Prometheus Adapter component for core platform monitoring is deprecated.
```
Comment (bburt-rh) [inline release_notes/ocp-4-15-release-notes.adoc:1441]:
```suggestion
With this release, the Prometheus Adapter component for core platform monitoring is deprecated and is planned to be removed in a future release. Red{nbsp}Hat will provide bug fixes and support for this component during the current release lifecycle, but this component will no longer receive enhancements and will be removed.
```

File: release_notes/ocp-4-15-release-notes.adoc [hunk 2]
```diff
@@ -1431,6 +1436,13 @@ The ability to enable dedicated service monitors by configuring the `dedicatedSe
 To replace this feature, Prometheus functionality has been improved to ensure that alerts and time aggregations are accurate.
 This improved functionality is active by default and makes the dedicated service monitors feature obsolete.
 
+[id="ocp-4-15-prometheus-adapter-for-core-platform-monitoring"]
+==== Prometheus Adapter for core platform monitoring
+With this release, the Prometheus Adapter component for core platform monitoring is deprecated. 
+The component is replaced by Metrics Server to ensure a more simplified and lightweight solution, because Metrics Server does not rely on Prometheus for its functionality. 
+Metrics Server also ensures scalability and a more accurate tracking of resource metrics. 
+This improved functionality is active by default if you enable the `TechPreviewNoUpgrade` option in the `FeatureGate` custom resource.
```
Comment (bburt-rh) [inline release_notes/ocp-4-15-release-notes.adoc:1444]:
```suggestion
With this release, the improved functionality of Metrics Server is available by default if you enable the `TechPreviewNoUpgrade` option in the `FeatureGate` custom resource.
```

**Comment by bburt-rh** [comment]
lgtm (as Monitoring DPM)

File: release_notes/ocp-4-15-release-notes.adoc [hunk 3]
```diff
@@ -1213,6 +1213,11 @@ In the following tables, features are marked with the following statuses:
 |General Availability
 |Deprecated
 
+|`prometheus-adapter` component thet collects resource metrics from Prometheus and exposes them in the metrics API.
```
Comment (simonpasquier) [inline release_notes/ocp-4-15-release-notes.adoc:1216]:
```suggestion
|`prometheus-adapter` component that queries resource metrics from Prometheus and exposes them in the metrics API.
```

**Comment by juzhao** [comment]
the only concern is if we can make metrics-server GA in 4.16, the 4.16 target branch date is May 17, 2024, but the epic is still in progress(bug on SNO cluster is not fixed), see [MON-3537](https://issues.redhat.com/browse/MON-3537). cc: @slashpai

**Review by GroceryBoyJr — COMMENTED** [review]
LGTM

**Pull request merged**